ray-operator/apis/ray/v1/raycluster_webhook.go

```diff
@@ -87,3 +92,26 @@ func (r *RayCluster) validateWorkerGroups() *field.Error {
 
 	return nil
 }
+
+func (r *RayCluster) ValidateRayClusterSpec() *field.Error {
+	if r.Annotations[RayFTEnabledAnnotationKey] == "false" && r.Spec.GcsFaultToleranceOptions != nil {
+		return field.Invalid(
+			field.NewPath("spec").Child("gcsFaultToleranceOptions"),
+			r.Spec.GcsFaultToleranceOptions,
+			fmt.Sprintf("GcsFaultToleranceOptions should be nil when %s is disabled", RayFTEnabledAnnotationKey),
+		)
+	}
+	if r.Annotations[RayFTEnabledAnnotationKey] != "true" &&
+		len(r.Spec.HeadGroupSpec.Template.Spec.Containers) > 0 &&
+		r.Spec.HeadGroupSpec.Template.Spec.Containers[RayContainerIndex].Env != nil {
```

Comment:
`r.Spec.HeadGroupSpec.Template.Spec.Containers[RayContainerIndex].Env != nil` check is not necessary.